server/util-server.js

```diff
@@ -254,6 +256,32 @@ exports.mssqlQuery = function (connectionString, query) {
     });
 };
 
+/**
+ * Check domain expiry date for the specified domain
+ * @param {string} hostname Hostname / address of machine
+ * @returns {Promise<Date>} Date Object on domain expiry
+ */
+exports.whoisExpiryDate = async function (domain) {
+    const whois = await WhoisLight.lookup({ format: true }, domain);
+
+    const expiryDates = Object.entries(whois).filter(([ key, value ]) => key.match(/(Expiry|Renewal) Date$/i)).map(([ key, value ]) => value);
```

Comment:
Allot more different phases are used, example for my personal use:

```suggestion
    const expiryDates = Object.entries(whois)
        .filter(([key, value]) => key.match(/(paid-till|valid|validity|exp|expiration|expiry|renewal|expires|expire)\s?(until|date|time|on)?$/i))
        .map(([key, value]) => value);
````